server/test/controllers/admin/AdminProgramControllerTest.java

```diff
@@ -174,8 +174,25 @@ public void edit_returnsExpectedForm() throws Exception {
     Result result = controller.edit(request, program.id);
 
     assertThat(result.status()).isEqualTo(OK);
+    assertThat(contentAsString(result)).contains("test program");
+
     assertThat(contentAsString(result)).contains("Edit program");
+    assertThat(contentAsString(result)).contains(CSRF.getToken(request.asScala()).value());
+  }
+
+  @Test
+  public void view_returnsReadOnlyVersion() throws Exception {
+    Request request = addCSRFToken(Helpers.fakeRequest()).build();
+    Program program = ProgramBuilder.newActiveProgram("test program").build();
+
+    Result result = controller.view(request, program.id);
+
+    assertThat(result.status()).isEqualTo(OK);
     assertThat(contentAsString(result)).contains("test program");
+
+    // TODO(jhummel) uncomment when the readOnlyView is used before submitting.
```

Comment:
This comment needs to be removed before submitting and after the other pull request is submitted.

Reply:
Remove extra blank line then too.